src/snowflake/snowpark/functions.py

```diff
@@ -684,6 +684,31 @@ def bitshiftleft(
     return call_builtin("bitshiftleft", c, n, _emit_ast=_emit_ast)
 
 
+@publicapi
+def bitshiftright_unsigned(
+    to_shift_column: ColumnOrName, n: Union[Column, int], _emit_ast: bool = True
+) -> Column:
+    """Returns the bitwise negation of a numeric expression.
+
+    Example:
+        >>> df = session.createDataFrame([(-1999)], ['a'])
+        >>> df.select(bitshiftright_unsigned('a', 1)).collect()[0][0]
+        9223372036854774808
+
+        >>> df = session.createDataFrame([(42)], ['a'])
+        >>> df.select(bitshiftright_unsigned('a', 1)).collect()[0][0]
+        21
+
+        >>> df = session.createDataFrame([(-21)], ['a'])
+        >>> df.select(bitshiftright_unsigned('a', 1)).collect()[0][0]
+        9223372036854775797
+    """
+    c = _to_col_if_str(to_shift_column, "bitshiftright_unsigned")
+    max_bit = bitshiftleft(lit(1), 64)
+    unsigned_c = iff(c < 0, bitshiftright(c + max_bit, n), bitshiftright(c, n))
+    return call_builtin("bitand", unsigned_c, max_bit - 1, _emit_ast=_emit_ast)
```

Comment:
```suggestion
    max_bit = bitshiftleft(lit(1, _emit_ast=False), 64, _emit_ast=False)
    unsigned_c = iff(c < 0, bitshiftright(c + max_bit, n, _emit_ast=False), bitshiftright(c, n, _emit_ast=False), _emit_ast=False)
    return call_builtin("bitand", unsigned_c, max_bit - 1, _emit_ast=_emit_ast)
```
let's make sure that for all internal calls to our functions, we set `_emit_ast=False`